### PR TITLE
Phrase share button feature.

### LINKED ISSUE #861: Make current phrase shareable
I would like to be able to  share the current phrase,  for example to send it via email or copy it into a word processor.

I could imagine multiple ways to support this:
- copy to clipboard by  ctrl+c
- add a copy to clipboard symbol next to "clear" and "delete"
- support dragging of the phrase to some other programm
- add a "share" button to select further options (at best user configurable providers)

### PR DESCRIPTION
**Pull request by tomivm**
This PR implements an optional Share button on the Output component. This allows sharing on social media the labels of the symbols on the 'output' like a Phrase. You can activate this on Settings/Navigation and Buttons.
image
image

### The problem right now is that this functionality is mainly to share URL content. This causes that Facebook and Reddit require a URL like a prop. We can pass the Cboard.io link but the phrase is shared only like a quote and no like a comment.
image
sorry that I have my Facebook in Spanish.

### CODE REVIEW

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      color="inherit"
+      onClick={onShareClick}
+      style={style}
+      hidden={hidden}
+    />
+    <Dialog
+      open={open}
+      onClose={onShareClose}
+      fullScreen={fullScreen}
+      className="ShareDialog__container"
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.component.js]: classname collides with the board share component

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        <IconButton
+          label={intl.formatMessage(messages.close)}
+          onClick={onShareClose}
+        >
+          <CloseIcon />
+        </IconButton>
+      </DialogTitle>
+      <DialogContent className="ShareDialog__content">
+        <div className="ShareDialog__Subtitle">
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.component.js]: no need for subtitle here, please remove element

File: src/components/Board/Output/PhraseShare/PhraseShare.css [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+.ShareDialog__container .ShareDialog__title button {
+  position: absolute;
+  top: 7px;
+  right: 7px;
+  color: #000;
+}
+
+.ShareDialog__container .ShareDialog__Subtitle {
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.css]: no need for subtitle here, please remove element

File: src/components/Board/Output/PhraseShare/PhraseShare.messages.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+export default defineMessages({
+  title: {
+    id: 'cboard.components.PhraseShare.title',
+    defaultMessage: 'Share or copy your Phrase'
+  },
+  close: {
+    id: 'cboard.components.PhraseShare.close',
+    defaultMessage: 'Close'
+  },
+  subtitle:{
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.messages.js]: no need for subtitle here, please remove element

File: src/translations/src/cboard.json [hunk 5]
```diff
@@ -135,6 +135,15 @@
   "cboard.components.BoardShare.reddit": "Reddit",
   "cboard.components.BoardShare.subject": "Check out this board from Cboard!",
   "cboard.components.BoardShare.body": "Hello! I want to share a communication board from the Cboard tool. Please find it at: {url}",
+  "cboard.components.PhraseShare.title": "Share or copy your Phrase",
+  "cboard.components.PhraseShare.subtitle": "Click and share!",
```
Comment (martinbedouret) [inline src/translations/src/cboard.json]: no need for subtitle here, please remove element

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+              <div>
+                <CopyIcon />
+              </div>
+              <FormattedMessage {...messages.copyLink} />
+            </div>
+          </Button>
+          <Button>
+            <EmailShareButton
+              subject={phrase}
+              body={phrase}
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.component.js:79]: email subject should not be the phrase itself, use the share board as an example

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+              onShareWindowClose ={onShareClose}
+            >
+              <FacebookIcon round />
+              <FormattedMessage id="facebook" {...messages.facebook} />
+            </FacebookShareButton>
+          </Button>
+          <Button>
+            <TwitterShareButton
+              //title={phrase}
+              //hashtags={['cboard', 'AAC']}
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.component.js]: test if we can include those hashtags

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+      </DialogContent>
+    </Dialog>
+  </React.Fragment>
+);
+
+PhraseShare.defaultProps = {
+  open: false,
+  disabled: false,
+  onShareClose: () => {},
+  copyLinkAction: () => {}
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.component.js]: the right name is `onCopyPhrase`

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+};
+
+PhraseShare.propTypes = {
+  open: PropTypes.bool,
+  phrase: PropTypes.string.isRequired,
+  onShareClose: PropTypes.func,
+  onShareClick: PropTypes.func.isRequired,
+  copyLinkAction: PropTypes.func
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.component.js]: the right name is `onCopyPhrase`

File: src/components/Board/Output/SymbolOutput/SymbolOutput.js [hunk 10]
```diff
@@ -86,10 +108,22 @@ class SymbolOutput extends PureComponent {
...
+          //   style={copyButtonStyle}
+          //   hidden={!symbols.length}
+          // />
+          <PhraseShare
+            label={intl.formatMessage(messages.share)}
+            intl={this.props.intl}
+            onShareClick={this.onShareClick}
+            onShareClose={this.onShareClose}
+            publishBoard={this.publishBoard}
+            copyLinkAction={onCopyClick}
```
Comment (martinbedouret) [inline src/components/Board/Output/SymbolOutput/SymbolOutput.js]: the right name is `onCopyPhrase`

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+
+const PhraseShare = ({
+  label,
+  phrase,
+  intl,
+  open,
+  fullScreen,
+  onShareClick,
+  onShareClose,
+  copyLinkAction,
```
Comment (martinbedouret) [inline src/components/Board/Output/PhraseShare/PhraseShare.component.js]: the right name is `onCopyPhrase`

File: src/components/Board/Output/SymbolOutput/SymbolOutput.js [hunk 12]
```diff
@@ -7,11 +7,30 @@ import IconButton from '@material-ui/core/IconButton';
 import Symbol from '../../Symbol';
 import BackspaceButton from './BackspaceButton';
 import ClearButton from './ClearButton';
-import CopyButton from './CopyButton';
+//import CopyButton from './CopyButton';
```
Comment (martinbedouret) [inline src/components/Board/Output/SymbolOutput/SymbolOutput.js]: remove

**Review by martinbedouret — CHANGES_REQUESTED** [review]

File: src/components/Board/Output/Output.container.js [hunk 13]
```diff
@@ -182,6 +182,11 @@ export class OutputContainer extends Component {
     this.clearOutput();
   };
 
+  handleGetPhraseToShare = () => {
+    const labels = this.props.output.map(symbol => symbol.label);
```
[the remark this answers is not in the document]
Comment (tomivm) [reply src/components/Board/Output/Output.container.js]: Thanks Martin! do you mean that we should add a condition on `handleCopyClick` too? to fix it on another PR.

File: src/components/Board/Output/PhraseShare/PhraseShare.component.js (hunk 7 above)
Comment (tomivm) [reply src/components/Board/Output/PhraseShare/PhraseShare.component.js]: yes we can! we can use `via` prop to tag cboard_io too. but I didn't know if we wanted that.

**Review by martinbedouret — APPROVED** [review]

**Pull request merged**